### PR TITLE
Abilities proposal markdown

### PR DESCRIPTION
**Pull request by JTeeuwissen**
Tried to be consistent with the google docs, took some creative liberty with problem references.
Not sure if the right directory, feel free to correct.

### CODE REVIEW

File: design/Abilities.md [hunk 1]
```diff
@@ -0,0 +1,545 @@
+# Proposal: Abilities in Roc
```
Comment (ayazhafiz) [inline design/Abilities.md:1]: What do we think of adding a "status" here, to make it clear how (im)mature the implementation is? We can also link to implementation design from there if we think it's appropriate.

Comment (JTeeuwissen) [reply design/Abilities.md:1]: Perhaps use multiple (checked) checkboxes to indicate progress and what's left to do.

Comment (ayazhafiz) [reply design/Abilities.md:1]: Personally, I feel that is better left to issue tracking - we already have that at https://github.com/roc-lang/roc/issues/2463. Maybe just something short that like "Status: we invite you to try out abilities for beta use, and are working on resolving known limitations"?

**Review by Anton-4 — APPROVED** [review]
Thanks @JTeeuwissen!

**Pull request merged**